### DISCUSSION TITLE
got `fatal: not a git repository (or any of the parent directories): .git` when `znap pull`

### DISCUSSION THREAD

**Discussion by tshu-w** [marlonrichert/zsh-snap#170] [Q&A]
Hi, I got `fatal: not a git repository (or any of the parent directories): .git` when `znap pull`. I tried to delete all plugins in `~znap` and reinstall them, but the error is still there.

**Comment by marlonrichert** [comment]
It appears you have a `.git` dir inside your plugins dir. Why?

**Reply by tshu-w** [reply]
I don't quite understand you. Isn't the plugin cloned, so there would be .git? If the plugins dir refers to `~znap` then there is no.
```console
λ cd ~znap && ls -al
total 136
drwxr-xr-x 14 wangtianshu staff    448 Mar 30 09:35 .
drwxr-xr-x 12 wangtianshu staff    384 Feb 23 15:57 ..
-rw-r--r--  1 wangtianshu staff   6148 Mar 30 09:38 .DS_Store
drwxr-xr-x  7 wangtianshu staff    224 Mar 30 09:35 conda-zsh-completion
-rw-------  1 wangtianshu staff 130528 Mar 30 09:39 history
drwx------  3 wangtianshu staff     96 Mar 30 09:38 site-functions
drwxr-xr-x 12 wangtianshu staff    384 Mar 30 09:37 wakatime-zsh-plugin
drwxr-xr-x 15 wangtianshu staff    480 Mar 30 09:37 zsh-autocomplete
drwxr-xr-x 26 wangtianshu staff    832 Mar 30 09:37 zsh-autosuggestions
drwxr-xr-x 12 wangtianshu staff    384 Mar 30 09:35 zsh-completions
drwxr-xr-x 12 wangtianshu staff    384 Mar 30 09:37 zsh-edit
drwxr-xr-x  7 wangtianshu staff    224 Mar 30 09:37 zsh-extract
drwxr-xr-x 15 wangtianshu staff    480 Mar 30 09:38 zsh-snap
drwxr-xr-x 22 wangtianshu staff    704 Mar 30 09:37 zsh-syntax-highlighting
```

**Reply by marlonrichert** [reply]
Sorry, no idea. I'm unable to reproduce this. It looks like `znap pull` is for some reason trying to run `git pull` inside a `.git` dir, but I have no idea why.

After doing `znap pull`, do `functions -T .znap.pull` and then run `znap pull` again. It should give you an idea of where it fails.

**Reply by tshu-w** [reply]
Thank you! I found out that it was because I turned off git-maintenance through `zstyle ':znap:*:*' git-maintenance off` and znap tried to unregister `~znap/site-functions`. Should we do something special with this directory on the znap side?

```
+.znap.pull:23> git -C /Users/wangtianshu/.local/share/zsh/site-functions maintenance unregister --quiet
fatal: not a git repository (or any of the parent directories): .git
```

**Reply by marlonrichert** [reply]
Znap is not expecting there to be anything else than repos in your plugins dir. You'll need to store your plugins somewhere else.

**Reply by tshu-w** [reply]
Thanks, I see.
